[IsDeprecatedWeakRefSmartPointerException] Make WebDeviceOrientationAndMotionAccessController ref-counted

RupinMittal opened the pull request:
#### 09fcd1e009e755c345383b09640e4bc2903bc54e
<pre>
[IsDeprecatedWeakRefSmartPointerException] Make WebDeviceOrientationAndMotionAccessController ref-counted
<a href="https://bugs.webkit.org/show_bug.cgi?id=281353">https://bugs.webkit.org/show_bug.cgi?id=281353</a>
<a href="https://rdar.apple.com/137781505">rdar://137781505</a>

Reviewed by Chris Dumez and Geoffrey Garen.

We forward WebDeviceOrientationAndMotionAccessController&apos;s ref/deref
to it&apos;s owner WebsiteDataStore.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::receivedNavigationActionPolicyDecision):
(WebKit::WebPageProxy::shouldAllowDeviceOrientationAndMotionAccess):
* Source/WebKit/UIProcess/WebsiteData/WebDeviceOrientationAndMotionAccessController.cpp:
(WebKit::WebDeviceOrientationAndMotionAccessController::WebDeviceOrientationAndMotionAccessController):
(WebKit::WebDeviceOrientationAndMotionAccessController::ref const):
(WebKit::WebDeviceOrientationAndMotionAccessController::deref const):
(WebKit::WebDeviceOrientationAndMotionAccessController::clearPermissions): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebDeviceOrientationAndMotionAccessController.h:
(WebKit::WebDeviceOrientationAndMotionAccessController::clearPermissions):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::WebsiteDataStore):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
(WebKit::WebsiteDataStore::protectedDeviceOrientationAndMotionAccessController):

Canonical link: <a href="https://commits.webkit.org/285242@main">https://commits.webkit.org/285242@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e708dcbbcdb44a85271d84e45e0a7b58a6d63173

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71900 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51320 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24685 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76060 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23109 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59121 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22929 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56749 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15255 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74967 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46548 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61941 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37187 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43211 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21454 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65123 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19775 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77740 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16140 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18962 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/64886 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16186 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61964 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64485 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12664 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6302 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11045 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47118 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1902 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48187 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49474 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47931 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
r=me

cdumez reviewed (approved):
Seems OK though we could have also forwarded the ref-counting to the owner.

RupinMittal committed:
[IsDeprecatedWeakRefSmartPointerException] Make WebDeviceOrientationAndMotionAccessController ref-counted
https://bugs.webkit.org/show_bug.cgi?id=281353
rdar://137781505

Reviewed by Chris Dumez and Geoffrey Garen.

We forward WebDeviceOrientationAndMotionAccessController's ref/deref
to it's owner WebsiteDataStore.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::receivedNavigationActionPolicyDecision):
(WebKit::WebPageProxy::shouldAllowDeviceOrientationAndMotionAccess):
* Source/WebKit/UIProcess/WebsiteData/WebDeviceOrientationAndMotionAccessController.cpp:
(WebKit::WebDeviceOrientationAndMotionAccessController::WebDeviceOrientationAndMotionAccessController):
(WebKit::WebDeviceOrientationAndMotionAccessController::ref const):
(WebKit::WebDeviceOrientationAndMotionAccessController::deref const):
(WebKit::WebDeviceOrientationAndMotionAccessController::clearPermissions): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebDeviceOrientationAndMotionAccessController.h:
(WebKit::WebDeviceOrientationAndMotionAccessController::clearPermissions):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::WebsiteDataStore):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
(WebKit::WebsiteDataStore::protectedDeviceOrientationAndMotionAccessController):

Canonical link: https://commits.webkit.org/285242@main
